tests/compilation/test_predictor_rl.py

```
@@ -136,3 +141,38 @@ def test_register_action() -> None:
+    val, kind = predictor.env.calculate_reward(qc=qc, mode="auto")
+    assert kind == "approx"
+    assert 0.0 <= val <= 1.0
+
+    # Ensure caching produced per-gate mappings
+    assert predictor.env._dev_avgs_cached  # noqa: SLF001
+    assert isinstance(predictor.env._err_by_gate, dict)  # noqa: SLF001
+    assert isinstance(predictor.env._dur_by_gate, dict)  # noqa: SLF001
+    assert len(predictor.env._err_by_gate) > 0  # noqa: SLF001
+
+    if figure_of_merit == "estimated_success_probability":
+        assert len(predictor.env._dur_by_gate) > 0  # noqa: SLF001
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (non-enabled: `SLF001`)

Remove unused `noqa` directive

(RUF100)